includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -485,11 +490,50 @@ public function process_payment_for_order( $args ) {
 
 			$intent_id = $intent->get_id();
 			$status    = $intent->get_status();
+			$charge_id = $intent->get_charge_id();
+		} else {
+			// For $0 orders, we need to save the payment method using a setup intent.
+			$intent = $this->payments_api_client->create_setup_intent(
+				$payment_method,
+				$customer_id,
+				'true'
+			);
+
+			$intent_id = $intent['id'];
+			$status    = $intent['status'];
+			$charge_id = '';
 
+			// In SCA cases the setup intent status might be requires_action and we should display the authentication modal.
+			// For now, since we're not supporting SCA cards, we can ignore that status.
+			if ( 'succeeded' !== $status ) {
+				throw new Exception( __( 'Failed to add the provided payment method. Please try again later', 'woocommerce-payments' ) );
```

Comment:
When we get to implement SCA support, we might be able to remove this exception and reuse the regular checkout code.

Reply:
Linked this note form the SCA issue https://github.com/Automattic/woocommerce-payments/issues/833